scripts/dts/edtlib.py

```diff
@@ -416,14 +416,18 @@ def _check_binding(self, binding, binding_path):
                 _err("missing, malformed, or empty '{}' in {}"
                      .format(prop, binding_path))
 
-        ok_top = {"title", "description", "compatible", "properties", "#cells",
-                  "parent-bus", "child-bus", "parent", "child",
-                  "child-binding", "sub-node"}
+        ok_top = {"title", "description", "compatible", "variant-compatible",
+                  "properties", "#cells", "parent-bus", "child-bus", "parent",
+                  "child", "child-binding", "sub-node"}
 
         for prop in binding:
             if prop not in ok_top and not prop.endswith("-cells"):
                 _err("unknown key '{}' in {}, expected one of {}, or *-cells"
                      .format(prop, binding_path, ", ".join(ok_top)))
+            if prop == "variant-compatible":
+                if not all(isinstance(elm, str) for elm in binding[prop]):
+                    _err("all elements in '{}:' in {} should be strings"
+                         .format(prop, binding_path))
```

Comment:
Should verify that it's a list as well, or it'll pass for `variant-compatible: "foo"` by iterating over "f", "o", "o".

Reply:
Could maybe put it outside the loop. Think some other stuff was:

```python
if "variant-compatible" in binding:
    var_comp = binding["variant_compatible"]
    if not (isinstance(var_comp, list) and
            all(isinstance(elm, str) for elm in var_comp)):
        _err("... should be a list of strings")
```

Reply:
Could put it next to the

```python
if "child-binding" in binding:
    ....

if "sub-node" in binding:
    ...

if "#cells" in binding:
    ...
```

stuff below.